### PR TITLE
Always collect AbstractRange

### PR DESCRIPTION
**Pull request by bkamins**
Following #1388 this PR always collects `AbstractRange` before assignment to a `DataFrame`. In this way we have consistency between constructor and `setindex!`.

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan any opinion on this?

File: src/dataframe/dataframe.jl [hunk 1]
```diff
@@ -43,6 +43,8 @@ A `DataFrame` is a lightweight object. As long as columns are not
 manipulated, creation of a DataFrame from existing AbstractVectors is
 inexpensive. For example, indexing on columns is inexpensive, but
 indexing by rows is expensive because copies are made of each column.
+An exception is assignment of `AbstractRange` as a column of
```
Comment (nalimilan) [inline src/dataframe/dataframe.jl]: I don't think "exception" is correct, as the previous sentence does not refer to columns passed to the constructor. Better add a separate paragraph, which would also document that column vectors are not copied in general.

I guess `setindex!` should be changed too, and the docstring be adapted?

Comment (bkamins) [reply src/dataframe/dataframe.jl]: Second try to clarify things.

`setindex!` was changed already (but no `setindex!` methods needed changing directly). It does not have its docstring so I have left it as is for now (hopefully in the near future I will have time to go back to #1380 as it is a major redesign and then I will write an updated docstring).

Comment (nalimilan) [reply src/dataframe/dataframe.jl]: Has it? I see this:
```julia
julia> x = DataFrame(x=1:2)
2×1 DataFrames.DataFrame
│ Row │ x │
├─────┼───┤
│ 1   │ 1 │
│ 2   │ 2 │

julia> x[:x] = 2:3
2:3

julia> x[:x]
2:3

```

Comment (bkamins) [reply src/dataframe/dataframe.jl]: Try testing with this PR checked out. You will see:
```
julia> x[:x] = 2:3
2:3

julia> x[:x]
2-element Array{Int64,1}:
 2
 3
```

Comment (nalimilan) [reply src/dataframe/dataframe.jl]: OK, that's what I realized after posting. But then it should be tested, right?

Comment (bkamins) [reply src/dataframe/dataframe.jl]: This is exactly what lines 624 and 625 test.

Comment (nalimilan) [reply src/dataframe/dataframe.jl]: Damn, let's say it was too late in the evening... :-/

**Comment by bkamins** [comment]
@nalimilan thanks!

**Pull request merged**